cogs/Moderation.py

```diff
@@ -253,6 +267,25 @@ async def check_multipost(self, message: discord.Message) -> None:
     async def on_message(self, message: discord.Message) -> None:
         await self.check_multipost(message)
 
+    @commands.Cog.listener()
+    async def on_raw_message_delete(self, payload: discord.RawMessageDeleteEvent):
+        if payload.message_id in self.multipost_warnings:
+            # The person deleted their message after seeing out multipost warning
+            # so we can delete the warning message
+            warning_message = self.multipost_warnings.pop(payload.message_id)
+            await warning_message.delete()
+
+        # If you delete your message then re-send it in another channel, that's fine
+        # so we can remove the original message fingerprint
+        for i, fingerprint in enumerate(self.fingerprints):
+            if (
+                payload.message_id == fingerprint.message_id
+                and payload.channel_id == fingerprint.channel_id
+                and payload.guild_id == fingerprint.guild_id
+            ):
+                del self.fingerprints[i]
+                break
```

Comment:
p.s. this also fixes the issues with google surveys, where we auto-delete their message and suggest that they post in #surveys instead, where they then copy/paste the same message into #surveys and we detect a multipost (even though the original message was deleted)